util/generate_testrun_index.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/python3
+
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from google.cloud import storage
+
+"""
+Scans all WPT results directories and generates an index.
+
+You must be logged into gcloud to run this script. The non-public permission
+necessary to generate this index is `bucket.list_blobs`.
+"""
+
+GCP_PROJECT = 'wptdashboard'
+RESULTS_BUCKET = 'wptd'
+
+
+def main():
+    storage_client = storage.Client(project=GCP_PROJECT)
+    bucket = storage_client.get_bucket(RESULTS_BUCKET)
+    by_sha = {}
+    by_platform = {}
```

Comment:
For `by_sha` and `by_platform`, you could have a comment above explaining what these are for.

It looks like `by_sha` maps hashes (of what I'm not sure - wpt commits?) to lists of platforms; and `by_platform` maps platform strings to lists of hashes. These two things are serialized as JSON and uploaded as "testruns-index.json" (which will be overwritten/updated when this script is rerun?)